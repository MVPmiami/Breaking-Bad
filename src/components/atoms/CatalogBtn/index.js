import styles from "./style.module.scss";

const CatalogBtn = ({isActiveBtn}) => {
  return (
    <div className={`${styles.btnWrapper} ${isActiveBtn ? styles.active : styles.disable}`}>
      <div className={styles.burger}>
        <span></span>
      </div>
      <div className={styles.text}>Catalog</div>
    </div>
  );
};

export default CatalogBtn;
