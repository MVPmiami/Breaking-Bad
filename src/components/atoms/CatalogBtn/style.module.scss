@import "../../../assets/styles/variables.scss";

.btnWrapper {
  border: 0.087rem solid $CatalogBcgActive;
  background-color: $CatalogBcgDisable;
  box-sizing: border-box;
  border-radius: 0.25rem;
  width: 7.813rem;
  height: 2.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 1.3rem;
  &:hover {
    cursor: pointer;
  }
  .burger {
    display: block;
    position: relative;
    width: 1rem;
    height: 0.75rem;
    &::before,
    &::after {
      content: "";
      position: absolute;
      background: $dataText;
      width: 100%;
      height: 0.125rem;
    }
    &::before {
      top: 0;
    }
    &::after {
      bottom: 0;
    }
    & span {
      position: absolute;
      background: $dataText;
      left: 0;
      width: 100%;
      height: 0.125rem;
      top: 0.313rem;
    }
  }
  .text {
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    font-size: 1rem;
    line-height: 1.5rem;
    color: $dataText;
  }
}

.btnWrapper.active {
  background-color: $CatalogBcgActive;
}

.btnWrapper.disable {
  background-color: $CatalogBcgDisable;
}
