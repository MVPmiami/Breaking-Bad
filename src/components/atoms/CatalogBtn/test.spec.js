import React from "react";
import CatalogBtn from "../../../../src/components/atoms/CatalogBtn";

const setUp = (props) => shallow(<CatalogBtn {...props} />);

describe("should render CatalogBtn component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should contain .btnWrapper wrapper", () => {
    const wrapper = component.find(".btnWrapper");
    expect(wrapper.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it(".btnWrapper should contain .active class", () => {
    component = shallow(<CatalogBtn isActiveBtn={true} />);
    const wrapper = component.find(".active");
    expect(wrapper.length).toBe(1);
  });
});
