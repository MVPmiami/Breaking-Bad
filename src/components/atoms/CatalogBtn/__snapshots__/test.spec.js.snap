// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CatalogBtn component should contain .btnWrapper wrapper 1`] = `
<div
  className="btnWrapper disable"
  data-test-catalogbtn="catalog-btn"
>
  <div
    className="burger"
  >
    <span />
  </div>
  <div
    className="text"
  >
    Catalog
  </div>
</div>
`;
